### DISCUSSION TITLE
MooseFS on VDO?

### DISCUSSION THREAD

**Discussion by dragon2611** [moosefs/moosefs#337] [Q&A]
Has anyone else tried a mooseFS chunkserver on a VDO volume or am I the only one crazy enough?

No idea if VDO will be able to de-deduplicate the data once it's been through mooseFS but i guess we will see, Obviously if it works at all it would only deal with duplicate data on that chunk server.

**Comment by chogata** [comment]
We have no practical experience with VDO, so if you try this, please, share your experience.

One thing you might need to remember is the `~` option for a disk in `mfshdd.cfg`. A chunk server checks the total number of blocks on each disk periodically and if it detects a significant change, it will mark the disk as damaged. Sometimes in compressed filesystems this change is big enough to trigger this mechanism, so if you want to avoid that, adding `~` in front of disks with compressed filesystems will make the chunk server ignore this check for them. Now, I don't know how a VDO volume will report the total number of blocks, but this is something to keep in mind.

**Comment by dragon2611** [comment]
I don't think moosefs will be able to see what VDO is doing as VDO is sitting beneath the file-system (EXT4) in this case, essentially how you expose the de-duplication is by creating a logical volume that is  bigger than the physical underlaying volume.

At the moment I've made the logical volume only slightly bigger than the physical to ensure I don't run out of space on the underlying block device.  It seems that the de-duplication does work even with mooseFS chunks but as expected the ratios largely depend on what is being stored on the volume.

**Comment by chogata** [comment]
> essentially how you expose the de-duplication is by creating a logical volume that is bigger than the physical underlaying volume.

And that is exactly what might be the "problem", unless the logical volume's size doesn't change over time. If it does, then the `~` option might become necessary. Other than that, I don't see a reason for it not to work. How efficiently will it work is another question, but hopefully your tests will turn out positive in that aspect.

**Comment by dragon2611** [comment]
Seems to be working thus far other than iowait when moving a lot data, but that's expected given one of my volumes is on (very) slow storage.

A create coal targeting the SSDs then a keep targeting the VDO seems to be working well for me so far.

**Comment by jwest75674** [comment]
+1 for VDO from me. I've been running is on that other liz file system for a few months. Recently switched to MooseFS and VDO hasn't been an issue on either.
